main.cpp

```diff
@@ -197,6 +197,8 @@ void thread_complete(){
 
     int ret_val = sigsetjmp(threads[current_thread_id].env,1);
     if (ret_val == 1) {
+	// Reset the quantum
+	setitimer(ITIMER_VIRTUAL, &timer, nullptr);
 	enable_interrupts();
```

Comment:
Resets the quantum right before the thread returns to where ever it was pre-empted before